src/helpers.rs

```diff
@@ -601,12 +599,10 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         os_str: &OsStr,
         memkind: MemoryKind<MiriMemoryKind>,
     ) -> MPlaceTy<'tcx, Tag> {
-        if cfg!(target_os = "unix") {
-            self.alloc_os_str_as_c_str(os_str, memkind)
-        } else if cfg!(target_os = "windows") {
-            self.alloc_os_str_as_wide_str(os_str, memkind)
-        } else {
-            panic!("support for target OS not yet available")
+        let target_os = self.eval_context_ref().tcx.sess.target.target.target_os.as_str();
+        match target_os {
+            "linux" => self.alloc_os_str_as_c_str(os_str, memkind),
+            "windows" => self.alloc_os_str_as_wide_str(os_str, memkind),
```

Comment:
This doesn't look like it compiles? The pattern is not exhaustive. Please at least build-test before pushing for review. :)